packages/dev/core/src/Misc/greasedLineTools.ts

```diff
@@ -524,8 +524,14 @@ export class GreasedLineTools {
      * @returns the colors texture
      */
     public static CreateColorsTexture(name: string, colors: Color3[], colorsSampling: number, scene: Scene) {
+        const maxTextureSize = scene?.getEngine()?.getCaps().maxTextureSize ?? 1;
+        const width = colors.length > maxTextureSize ? maxTextureSize : colors.length;
+        const height = Math.ceil(colors.length / maxTextureSize);
+        if (height > 1) {
+            colors = [...colors, ...Array(width * height - colors.length).fill(colors[0])];
+        }

```

Comment:
I don't think we need this?

Reply:
The idea is to fill the pixels of the texture (sampler) with arbitrary values such that it form a rectangle:
```
********
****----
```
 with `*` as colors element and `-` as filler color element.
 I get `WebGL: INVALID_OPERATION: texImage2D: ArrayBufferView not big enough for request` if i do not do this.

If we are able to create a texture with width smaller than the colors length, the texture can have height of 1 pixel (no extra filler needed), otherwise we need to fill it to satisfy the shader buffer size, e.g. the texture need to be a rectangle?

I can be mistaken, please guide me if i am wrong :)

Reply:
Yes, my bad, it's fine then.